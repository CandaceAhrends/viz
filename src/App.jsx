import React, { useEffect } from 'react';
import Navigation from './components/navigation/Navigation.jsx';
import Router from './components/content/Router.jsx';
import StockProvider from './StockContext.jsx';
import { setDate } from './features/stocksSlice';
import { getPreviousMarketDate } from './utils';
import { useAppDispatch } from './hooks';
import { BrowserRouter } from 'react-router-dom';
import './app.scss';

const loadPath = window.location.href?.split('/').pop() || '/';

function App() {
  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(setDate(getPreviousMarketDate()));
  }, []);

  return (
    <StockProvider>
      <BrowserRouter>
        <div className="flex w-screen h-screen">
          <Navigation loadPath={loadPath}></Navigation>
          <Router></Router>
        </div>
      </BrowserRouter>
    </StockProvider>
  );
}

export default App;
