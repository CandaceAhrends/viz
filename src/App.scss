body {
  background: black;

  .datepicker-container {
    .custom-calendar {
      background-color: #070807;
      border: none;
      border-radius: 10px;
      padding: 10px;
      color: rgb(12, 20, 16);
    }

    /* Day cell styling */
    .react-datepicker__day {
      color: #07f8b5;
      background-color: transparent;
      border-radius: 50%;
      display: inline-flex;
      justify-content: center;
      align-items: center;
      height: 2.7rem;
      width: 2.7rem;
      margin: 5px;
    }

    /* Selected day styling */
    .react-datepicker__day--selected {
      background-color: #00aaff;
      color: rgb(91, 112, 85);
    }

    /* Current day styling */
    .current-day {
      border: 2px solid #00aaff;
      border-radius: 50%;
    }

    /* Navigation buttons */
    .react-datepicker__navigation {
      background-color: transparent;
      border: none;
    }

    .react-datepicker__navigation-icon::before {
      border-color: rgb(36, 30, 30);
    }

    .react-datepicker__day-name {
      color: #323636;
    }

    .react-datepicker__day--selected {
      background-color: #00aaff;
      color: black;
    }
    .react-datepicker__month {
      border: solid #1e1c1c;
    }
    .react-datepicker-wrapper {
      color: #1b1919;
      .react-datepicker__view-calendar-icon input {
        padding: 0.3rem 2rem;
      }
    }

    .react-datepicker__navigation--next {
      right: 12px;
      height: 4rem;
    }
    .react-datepicker__navigation--previous {
      left: 12px;
      height: 4rem;
    }
    .react-datepicker__day--disabled,
    .react-datepicker__month-text--disabled,
    .react-datepicker__quarter-text--disabled,
    .react-datepicker__year-text--disabled {
      color: black;
    }
    .current-day {
      color: black;
      border: none;
    }
    .react-datepicker__day-name:first-child,
    .react-datepicker__day-name:last-child {
      display: none;
    }
    .react-datepicker__day-name,
    .react-datepicker__day,
    .react-datepicker__time-name {
      width: 2.5rem;
    }
  }
}
