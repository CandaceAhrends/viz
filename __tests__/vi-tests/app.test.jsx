import React from 'react';
import { describe, it, expect, vi } from 'vitest';
import ReactDOM from 'react-dom';
import { render, fireEvent, screen } from '@testing-library/react';

import App from '../../src/App';

describe('App', () => {
  it('renders App component', () => {
    const { container } = ReactDOM.render(<App />);
    expect(container).toMatchSnapshot();
  });
});
