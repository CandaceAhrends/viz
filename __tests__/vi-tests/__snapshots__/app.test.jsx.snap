// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders App component 1`] = `"App"`;
